kratos/geometries/geometry_data.h

```diff
@@ -769,46 +776,28 @@ class GeometryData
     ///@name Member Variables
     ///@{
 
-    SizeType mDimension;
+    GeometryDimension const* mpGeometryDimension;
 
-    SizeType mWorkingSpaceDimension;
-
-    SizeType mLocalSpaceDimension;
-
-    enum IntegrationMethod mDefaultMethod;
+    IntegrationMethod mDefaultMethod;
 
     IntegrationPointsContainerType mIntegrationPoints;
 
-    ShapeFunctionsValuesContainerType mShapeFunctionsValues;
-
-    ShapeFunctionsLocalGradientsContainerType mShapeFunctionsLocalGradients;
+    GeometryShapeFunctionContainer<IntegrationMethod> mGeometryShapeFunctionContainer;
 
     ///@}
     ///@name Serialization
     ///@{
+    template<typename TPointType>
+    friend class Geometry;
 
     friend class Serializer;
 
     virtual void save( Serializer& rSerializer ) const
```

Comment:
Empty serializer?

Reply:
before that, the serializer had only the dimensional data. This is now in the external class which has its own serializer. As this one is static byitself, I think it does not need to be serialized here.

Reply:
i guess you'll need a test for this...

Reply:
also if you don't call here the serialization of the GeoemtryDimension, it will not be called anywhere (i think)

Reply:
You should save and load the pointer to GeometryDimension here. Nevertheless in the goemetries with static data you won't save and load the GeometryData and will construct it again. So this method won't be called.

Definitely test is needed.